paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/PaymentSelectionUpdater.kt

```diff
@@ -85,5 +84,16 @@ internal class DefaultPaymentSelectionUpdater @Inject constructor(
     }
 }
 
+private fun PaymentSheet.Configuration?.containsViolateDifferences(
+    other: PaymentSheet.Configuration
+): Boolean {
+    if (this == null) {
+        return true
+    }
+
+    return customer != other.customer ||
+        billingDetailsCollectionConfiguration != other.billingDetailsCollectionConfiguration
```

Comment:
From what I can tell, these were the main items of concern when looking for configuration changes that affect all payment methods. Is there anything else we might be missing?

Reply:
It seems like we want basically all non UI things in this check. Even some of the google pay things (like changing from test to live mode) seems like we'd want to throw the old selection out.